src/nvim/lua/treesitter.c

```diff
@@ -1538,8 +1538,8 @@ int tslua_parse_query(lua_State *L)
   TSQuery *query = ts_query_new(lang, src, (uint32_t)len, &error_offset, &error_type);
 
   if (!query) {
-    return luaL_error(L, "query: %s at position %d for language %s",
-                      query_err_string(error_type), (int)error_offset, lang_name);
+    return luaL_error(L, "%s",
+                      query_err_string(src, error_offset, error_type));
```

Comment:
Could we in a follow-up PR forward a table to Lua to account for possible pre-processing of the query source? Lua could subtract lines that come from `; inherit` or forward to a different file in case the error is caused by a `inherit` directive?

Anyways not directly related to this PR... Great work!